scripts/basic-workflow-tests.sh

```diff
@@ -257,12 +258,23 @@ test_convert() {
 }
 
 test_evaluate() {
+    local MODEL_PATH="${DATA_HOME}/instructlab/models/instructlab/granite-7b-lab"
+    local BASE_MODEL_PATH="${DATA_HOME}/instructlab/models/instructlab/merlinite-7b-lab"
     task Evaluate the model
 
     if [ "$EVAL" -eq 1 ]; then
       export INSTRUCTLAB_EVAL_MMLU_MIN_TASKS=true
       export HF_DATASETS_TRUST_REMOTE_CODE=true
-      ilab model evaluate --model "${DATA_HOME}/instructlab/models/instructlab/granite-7b-lab" --benchmark mmlu
+      ilab model evaluate --model "${MODEL_PATH}" --benchmark mmlu
+      ilab model evaluate --model "${MODEL_PATH}" --benchmark mmlu_branch --sdg-path tests/testdata/mmlu_branch/ --base-model "${BASE_MODEL_PATH}"
+      export INSTRUCTLAB_EVAL_FIRST_N_QUESTIONS=10
+      ilab model evaluate --model "${MODEL_PATH}" --judge-model "${MODEL_PATH}" --benchmark mt_bench
```

Comment:
Are we running this with vllm or llama-cpp?

Reply:
Running with vllm
cc @alimaredia

Reply:
you sure? I don't think these jobs are set up to use vllm.  The script has some support, but it's not working in any CI working flow yet (that I know of).

Reply:
If it's not using vllm, mt_bench might be too slow.  Or we'll at least need to crank down first_n even lower.

Reply:
@russellb has you or anyone else tried the vLLM flags for this script yet? I don't think Alina has been able to run this script yet, could we do a pass to see what happens?

I know in future Seb and I would like to autodetect the backend based on the file passed to `--model` for every command. Unless `--backend` was applied here, I'd prefer to keep it that way.

Reply:
Even with autodetection, it may be good to do the explicit vLLM args here and have autodection be a separate unit/functional test. 

Are the necessary args implemented? If so we can implement/test them as part of the workflow in this PR IMHO. Assuming we can verify on one of @russellb's new AWS instance jobs before merging.

Reply:
@nathan-weinberg there is a `--backend vllm` flag but that does not work with CI currently

Reply:
Interesting - what is the root cause for that?

Reply:
Ali mentioned that the CI ami isn't able to run vLLM because of the cuda drivers

Reply:
@danmcp I cranked down `INSTRUCTLAB_EVAL_FIRST_N_QUESTIONS` from 10 to 5